### PR TITLE
Add Relaxed Data Binding Support for Client Generation

### PR DESCRIPTION
**Pull request by SachinAkash01**
Fixes: [#7413](https://github.com/ballerina-platform/ballerina-library/issues/7413)
See also: [#7366](https://github.com/ballerina-platform/ballerina-library/issues/7366)

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: openapi-cli/src/test/resources/expected_gen/default_headers_conflict_with_query.bal [hunk 1]
```diff
@@ -301,6 +301,10 @@ public type ConnectionConfig record {|
     http:ProxyConfig proxy?;
     # Enables the inbound payload validation functionality which provided by the constraint package. Enabled by default
     boolean validation = true;
+    # Enables or disables relaxed data binding on the client side. Disabled by default.
```
Comment (lnash94) [inline openapi-cli/src/test/resources/expected_gen/default_headers_conflict_with_query.bal:304]:
```suggestion
    # Enables or disables relaxed data binding on the client side. Enabled by default.
```

File: openapi-core/src/main/java/io/ballerina/openapi/core/generators/client/AuthConfigGeneratorImp.java [hunk 2]
```diff
@@ -1333,6 +1338,19 @@ private List<Node> getClientConfigRecordFields() {
                 validationMetadata, null, validationFieldType, validationFieldName,
                 equalToken, createRequiredExpressionNode(createToken(TRUE_KEYWORD)), semicolonToken);
         recordFieldNodes.add(validateFieldNode);
+
+        // add laxBinding for data binding
+        String apiComment = "Enables or disables relaxed data binding on the client side. " +
+                "Disabled by default.\nWhen enabled, the JSON data will be projected to the Ballerina record type" +
+                " and during the projection, \nnil values will be considered as optional fields and absent fields " +
+                "will be considered for nilable types";
```
Comment (lnash94) [inline openapi-core/src/main/java/io/ballerina/openapi/core/generators/client/AuthConfigGeneratorImp.java:1346]:
```suggestion
        String apiComment = "Enables relaxed data binding on the client side. When enabled, `nil` values are treated as optional, \nand absent fields are handled as `nilable` types. Enabled by default.";
```

Comment (TharmiganK) [reply openapi-core/src/main/java/io/ballerina/openapi/core/generators/client/AuthConfigGeneratorImp.java:1346]: +1

**Review by lnash94 — APPROVED** [review]
+1 to address review suggestions in separate PR

**Pull request merged**